### PR TITLE
se050: update SCP03 information and provisioning

### PR DESCRIPTION
**Pull request by ldts**
Updates to reflect the code changes.

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for bee7627e5802a02fe8ed01c3879730a786185627 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1779/docs/artifacts/html/index.html

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 1]
```diff
@@ -122,32 +122,38 @@ Secure Communication Protocol 03
...
-      We can choose whether to enable SCP03 right after boot with its default set of
-      keys or at a later time once the RPMB-FS (or REE-FS) are available so the
-      keys can be read from secure storage::
+       The user can select whether to enable SCP03 during secure world initialization
+       or at a later time from the normal world using the `scp03`_ command::
 
-	CFG_CORE_SE05X_SCP03_EARLY=y     : enables SCP03 with its default keys.
-	CFG_CORE_SE05X_SCP03_PROVISION=y : allows SCP03 rotation.
+	CFG_CORE_SE05X_SCP03_EARLY=y : enables SCP03 before the Normal World has booted.
+	CFG_SCP03_PTA=y              : allows SCP03 to be enabled from from the Normal World
```
Comment (MrCry0) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:133]: s/from from/from/

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 2]
```diff
@@ -122,32 +122,38 @@ Secure Communication Protocol 03
...
+OP-TEE from its core secret: the Hardware Unique Key (HUK)
 
 .. warning::
-     If the secure database storing the SCP03 keys gets corrupted, the processor will
-     no longer be able to access the SE05x over an encrypted connection. Moreover, there is
-     no protocol defined to recover from this situation. This is why
-     we chose to derive the SCP03 static keys from the platform's Hardware Unique
-     Key so that they can be recreated in the Trust Zone on every boot.
+     Once the static SCP03 keys have been derived from the HUK and programmed
+     into the device's NVM the **HUK MUST NOT CHANGE**.
```
Comment (MrCry0) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:145]: "...be changed"?

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 3]
```diff
@@ -204,18 +210,40 @@ database does not impose restrictions to the user.
           $ pkcs11-tool --module /usr/lib/libckteec.so.0.1 --keypairgen --key-type RSA:4096 --id 01 --token-label fio --pin 87654321 --label SE_7F000001 
 
 
-We developed a very simple PTA `SE05x Object Import Application`_ to interface
-to the TEE and gain access to the SE05x to obtain certificates. This application
-also includes a wrapper to import the keys using the mechanism described
-earlier.
+We have also developed a simple tool, the `SE05x Object Import Application`_,
+that interfaces to the TEE and gains access to the SE05x to import not only keys
+but also certificates.
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:215]:
```suggestion
We have also developed a tool, the `SE05x Object Import Application`_.
This tool interfaces with the TEE and gains access to the SE05x to import keys
 *and* certificates.
```

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 4]
```diff
@@ -204,18 +210,40 @@ database does not impose restrictions to the user.
...
 
-The *certificates* are retrieved in DER format using the import PTA and then
-written to the pkcs#11 token.
+The *certificates* are retrieved in DER format using the APDU interface
+presented by the driver and then written to the pkcs#11 token.
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:218]:
```suggestion
presented by the driver, and are then written to the pkcs#11 token.
```

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 5]
```diff
@@ -204,18 +210,40 @@ database does not impose restrictions to the user.
...
 
-.. code-block:: none
+The application uses the `libseteec`_ to send the APDUs to the secure element
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:220]:
```suggestion
The tool uses `libseteec`_ to send the APDUs to the secure element.
```

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 6]
```diff
@@ -204,18 +210,40 @@ database does not impose restrictions to the user.
...
+and `libckteec`_ to interface to the PKCS#11 implementation.
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:221]:
```suggestion
and `libckteec`_ to interface with the PKCS#11 implementation.
```

**Comment by doanac** [comment]
Docs for 4a5465f79cccf6c4fb60a8427f1504d74406882a are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1780/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for e5b696fdf768cfd969ca6abbcd15218a1ebacfb9 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1782/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 557c0aa88d2a26058fd05781952417120e43a3bd are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1783/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 5d5cc68c534112157cba27b503800e8d51769889 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1784/docs/artifacts/html/index.html

File: source/reference-manual/security/secure-elements/secure-element.050.rst (hunk 1 above)
Comment (ldts) [reply source/reference-manual/security/secure-elements/secure-element.050.rst]: thanks @MrCry0, just committed the same change as proposed by @kprosise

File: source/reference-manual/security/secure-elements/secure-element.050.rst (hunk 2 above)
Comment (ldts) [reply source/reference-manual/security/secure-elements/secure-element.050.rst:145]: I'd rather keep it as is...meaning that sometimes the HUK might change even though the user don't know it happened. Worded this way I am hoping clients get some awareness of how the HUK might change on their platforms.

File: source/reference-manual/security/secure-elements/secure-element.050.rst [hunk 7]
```diff
@@ -204,18 +210,40 @@ database does not impose restrictions to the user.
           $ pkcs11-tool --module /usr/lib/libckteec.so.0.1 --keypairgen --key-type RSA:4096 --id 01 --token-label fio --pin 87654321 --label SE_7F000001 
 
 
-We developed a very simple PTA `SE05x Object Import Application`_ to interface
-to the TEE and gain access to the SE05x to obtain certificates. This application
-also includes a wrapper to import the keys using the mechanism described
-earlier.
+We have also developed a tool, the `SE05x Object Import Application`_.
+This tool interfaces with the TEE and gains access to the SE05x to import keys
+ *and* certificates.
```
Comment (kprosise) [inline source/reference-manual/security/secure-elements/secure-element.050.rst:215]:
```suggestion
*and* certificates.
```
fix for indentation error.

**Review by kprosise — COMMENTED** [review]
made additional suggestion to fix the indentation issue I introduced. Otherwise it LGTM.

**Comment by doanac** [comment]
Docs for 4c511d903cb95baad8f02d035397ac45caa29817 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1791/docs/artifacts/html/index.html

**Pull request merged**